### PR TITLE
IGNITE-19436 Add "--add-opens=java.base/java.time=ALL-UNNAMED" to jvm…

### PR DESCRIPTION
**Pull request by liyuj**
… args

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by liyuj** [comment]
I have merged the latest code, please check again.

**Comment by liyuj** [comment]
The code submitted in IGNITE-20142 is incomplete, and the 
jvmdefaults.bat has not been fixed, so Windows environments may still 
report errors.

Because my computer is in a Linux environment, I did not verify the 
issue in Windows environment.

**Comment by liyuj** [comment]
CREATE TABLE City (

  ID INT,

  Name VARCHAR,

  CountryCode CHAR(3),

  District VARCHAR,

  Population INT,

  PRIMARY KEY (ID, CountryCode)

) WITH "template=partitioned, backups=1,affinityKey=CountryCode,CACHE_NAME=City, KEY_TYPE=CityKey, VALUE_TYPE=City";


select /*+ QUERY_ENGINE('calcite') */ count(*) from City;


then:



[2023-12-06T21:17:44,691][ERROR][client-connector-#84%01%][ClientListenerProcessor] Runtime error caught during grid runnable execution: GridWorker [name=message-received-notify, igniteInstanceName=01, finished=false, heartbeatTs=1701868663872, hashCode=1036118805, interrupted=false, runner=client-connector-#84%01%]
java.lang.ExceptionInInitializerError: null
    at org.apache.ignite.internal.processors.query.calcite.exec.rel.MemoryTrackingNode.<clinit>(MemoryTrackingNode.java:30) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.visit(LogicalRelImplementor.java:677) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.visit(LogicalRelImplementor.java:120) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.rel.agg.IgniteColocatedHashAggregate.accept(IgniteColocatedHashAggregate.java:64) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.visit(LogicalRelImplementor.java:835) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.visit(LogicalRelImplementor.java:845) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.visit(LogicalRelImplementor.java:217) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.visit(LogicalRelImplementor.java:120) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.rel.IgniteProject.accept(IgniteProject.java:89) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.visit(LogicalRelImplementor.java:835) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.LogicalRelImplementor.go(LogicalRelImplementor.java:850) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.ExecutionServiceImpl.mapAndExecutePlan(ExecutionServiceImpl.java:578) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.ExecutionServiceImpl.executePlan(ExecutionServiceImpl.java:478) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.CalciteQueryProcessor.lambda$parseAndProcessQuery$9(CalciteQueryProcessor.java:490) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.CalciteQueryProcessor.processQuery(CalciteQueryProcessor.java:525) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.CalciteQueryProcessor.parseAndProcessQuery(CalciteQueryProcessor.java:478) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.CalciteQueryProcessor.query(CalciteQueryProcessor.java:362) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.GridQueryProcessor$2.applyx(GridQueryProcessor.java:3102) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.GridQueryProcessor$2.applyx(GridQueryProcessor.java:3082) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.util.lang.IgniteOutClosureX.apply(IgniteOutClosureX.java:36) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.GridQueryProcessor.executeQuery(GridQueryProcessor.java:3817) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.GridQueryProcessor.lambda$querySqlFields$3(GridQueryProcessor.java:3128) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.GridQueryProcessor.executeQuerySafe(GridQueryProcessor.java:3256) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.GridQueryProcessor.querySqlFields(GridQueryProcessor.java:3078) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.GridQueryProcessor.querySqlFields(GridQueryProcessor.java:3037) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.odbc.jdbc.JdbcRequestHandler.querySqlFields(JdbcRequestHandler.java:773) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.odbc.jdbc.JdbcRequestHandler.executeQuery(JdbcRequestHandler.java:641) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.odbc.jdbc.JdbcRequestHandler.doHandle(JdbcRequestHandler.java:311) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.odbc.jdbc.JdbcRequestHandler.handle(JdbcRequestHandler.java:251) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.odbc.ClientListenerNioListener.onMessage(ClientListenerNioListener.java:209) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.odbc.ClientListenerNioListener.onMessage(ClientListenerNioListener.java:55) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.util.nio.GridNioFilterChain$TailFilter.onMessageReceived(GridNioFilterChain.java:279) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.util.nio.GridNioFilterAdapter.proceedMessageReceived(GridNioFilterAdapter.java:109) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.util.nio.GridNioAsyncNotifyFilter$3.body(GridNioAsyncNotifyFilter.java:97) ~[ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.util.worker.GridWorker.run(GridWorker.java:125) [ignite-core-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.util.worker.GridWorkerPool$1.run(GridWorkerPool.java:70) [ignite-core-2.15.0.jar:2.15.0]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
    at java.lang.Thread.run(Thread.java:840) [?:?]
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final int java.time.LocalDate.year accessible: module java.base does not "opens java.time" to unnamed module @687080dc
    at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354) ~[?:?]
    at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297) ~[?:?]
    at java.lang.reflect.Field.checkCanSetAccessible(Field.java:178) ~[?:?]
    at java.lang.reflect.Field.setAccessible(Field.java:172) ~[?:?]
    at org.apache.ignite.internal.processors.query.calcite.exec.tracker.ObjectSizeCalculator.classInfo(ObjectSizeCalculator.java:155) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.tracker.ObjectSizeCalculator.access$400(ObjectSizeCalculator.java:57) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.tracker.ObjectSizeCalculator$SizeCalculatorImpl.<init>(ObjectSizeCalculator.java:308) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.tracker.ObjectSizeCalculator$SizeCalculatorImpl.<init>(ObjectSizeCalculator.java:299) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.tracker.ObjectSizeCalculator.addSysClsSize(ObjectSizeCalculator.java:132) ~[ignite-calcite-2.15.0.jar:2.15.0]
    at org.apache.ignite.internal.processors.query.calcite.exec.tracker.ObjectSizeCalculator.<clinit>(ObjectSizeCalculator.java:105) ~[ignite-calcite-2.15.0.jar:2.15.0]

File: bin/include/jvmdefaults.bat [hunk 1]
```diff
@@ -89,6 +89,7 @@ if %java_version% GEQ 15 (
     --add-opens=java.base/java.lang.invoke=ALL-UNNAMED ^
     --add-opens=java.base/java.math=ALL-UNNAMED ^
     --add-opens=java.sql/java.sql=ALL-UNNAMED ^
+    --add-opens=java.sql/java.time=ALL-UNNAMED ^
```
Comment (NSAmelchev) [inline bin/include/jvmdefaults.bat:92]: There is no such module: `java.sql/java.time`

https://github.com/openjdk/jdk/blob/jdk-17%2B35/src/java.sql/share/classes/module-info.java

**Comment by liyuj** [comment]
Sorry, I made a mistake.
I have synchronized these two files: jvmdefaults.bat&java9.adoc

**Pull request merged**